src/connection.js

```diff
@@ -416,7 +416,9 @@ class Connection extends EventEmitter {
         if (this.config.options.rowCollectionOnDone) {
           this.request.rst.push(token.columns);
         }
-        return this.request.emit('row', token.columns);
+        if (!(this.state === this.STATE.SENT_ATTENTION && this.request.paused)) {
```

Comment:
Because it would change the current behaviour and break the test case connection-test / cancelRequest.

It's not only cancel(), it's also the timeout.

The current behaviour is that all the remaining rows are emitted until the server receives the ATTENTION message and stops sending rows.

And there is the effect that the server sends the rows it has buffered to far when it receives the ATTENTION message. In an interactive tool (like SQL Server Management Studio) the user would want to see the rows that have been produced so far up to the time when the user clicks the cancel button.